### PR TITLE
Pull over more shared methods

### PR DESCRIPTION
**Pull request by rmulhol**
- Reduce need for generic methods in custom transformers

### CODE REVIEW

File: pkg/shared/constants/event_signature_generator.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+func GetOverloadedFunctionSignature(rawAbi, name string, paramTypes []string) string {
+	result, err := FindSignatureInAbi(rawAbi, name, paramTypes)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func FindSignatureInAbi(rawAbi, name string, paramTypes []string) (string, error) {
```
Comment (paytonrules) [inline pkg/shared/constants/event_signature_generator.go:58]: There's a few public functions here that don't have tests. Are they tested in the transformers then?

**Comment by paytonrules** [comment]
"Reduce need for generic methods in custom transformers" - can you elaborate on what you mean here? Did this mean some code was deleted?

File: pkg/shared/constants/event_signature_generator.go (hunk 1 above)
Comment (rmulhol) [reply pkg/shared/constants/event_signature_generator.go:58]: I think so - I believe most of these used to be private functions that were implementation details of tested public functions (e.g. `BiteSignature()` is tested, and that calls `GetEventTopicZero(biteMethod())` => `getSolidityFunctionSignature(Cat100ABI(), "Bite")` => `getContractABI("MCD_CAT_1.0.0")`).

That said in this _particular_ case there are tests for this function

**Comment by rmulhol** [comment]
> "Reduce need for generic methods in custom transformers" - can you elaborate on what you mean here? Did this mean some code was deleted?

Sure, want me to update the commit? What this means is that the functions I'm pulling into this repo used to be duplicated in the custom transformers (so I added them here, have PRs to remove them from there, e.g. https://github.com/makerdao/vdb-mcd-transformers/pull/345 )

**Comment by paytonrules** [comment]
The note you just put is more than sufficient.

**Review by paytonrules — APPROVED** [review]

**Pull request merged**